### PR TITLE
karma: Canonicalize nicks

### PR DESCRIPTION
**Pull request by Miciah**
Canonicalize nicks before looking them up in the database.

### CODE REVIEW

File: pkg/gomr/utils.go [hunk 1]
```diff
@@ -69,6 +70,42 @@ func Match(line, regex string) bool {
...
+		case r >= 'a' && r <= 'z' || r >= '0' && r <= '9':
+			// Lowercase letters and numerals are fine.
+			return r
+		}
+		return -1
+	}, nick)
+
+	// Strip all invalid leading characters that are otherwise valid.
+	nick = strings.TrimLeftFunc(nick, func(r rune) bool {
+		return (r >= '0' && r <= '9') || r == '-'
```
Comment (tiwillia) [inline pkg/gomr/utils.go]: Numbers are invalid leading characters? What if I'm a big fan of 3scale++?

**Review by tiwillia — CHANGES_REQUESTED** [review]
Looks great other than where you strip leading numbers, I'm not sure thats appropriate.

File: pkg/gomr/utils.go (hunk 1 above)
Comment (Miciah) [reply pkg/gomr/utils.go]: https://tools.ietf.org/html/rfc2812#page-7 says it's invalid.  That doesn't mean no server allows it though.

Comment (Miciah) [reply pkg/gomr/utils.go]: Ohh, I see; we don't want to restrict this to IRC nicks.  I'll loosen the restrictions, maybe just drop control characters and convert to lowercase (per how IRC defines lowercase).

Comment (Miciah) [reply pkg/gomr/utils.go]: Fixed.

**Review by tiwillia — APPROVED** [review]
Thanks for this PR!

**Pull request merged**